cli/src/main/java/org/openjdk/skara/cli/GitProxy.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.skara.cli;
+
+import org.openjdk.skara.proxy.HttpProxy;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class GitProxy {
+    private static String gitConfig(String key) throws IOException, InterruptedException {
+        var pb = new ProcessBuilder("git", "config", key);
+        pb.redirectOutput(ProcessBuilder.Redirect.PIPE);
+        pb.redirectError(ProcessBuilder.Redirect.INHERIT);
+        var p = pb.start();
+        var bytes = p.getInputStream().readAllBytes();
+        var res = p.waitFor();
+        if (res == 0) {
+            return new String(bytes, StandardCharsets.UTF_8).trim();
+        }
+        return "";
+    }
+
+    private static String proxyEnvironmentVariable() {
+        for (var key : List.of("http_proxy", "https_proxy")) {
+            var proxy = System.getenv(key);
+            if (proxy == null) {
+                proxy = System.getenv(key.toUpperCase());
+            }
+            if (proxy != null) {
+                return proxy;
+            }
+        }
+        return "";
+    }
+
```

Comment:
It looks like both these methods are unused?

Reply:
Oops, yes :smile: They are remnants from a prototype, before I realized I could just utilize the `HttpProxy` class.